### PR TITLE
Add more end-to-end tests

### PR DESCRIPTION
**Pull request by barbarah**
These end-to-end tests test two services:

- Clerk, it tests the log-in state. Are the buttons visible when logged in? For this, I created a test user and organization. The tests will open the Create Community and Manage Members modal (components from Clerk) and check if they are open. I have not added tests for creating communities and adding members. But I do edit the existing test community.

- Planet scale: There are tests to create, update and delete lists. I remove test data before all relevant tests. I do this before and not after the tests because this is in Cypress's best practice guide: https://docs.cypress.io/guides/references/best-practices#Using-after-Or-afterEach-Hooks

# Tests updates
I have run the tests many times; some tests sometimes fail. After filtering, the tests don't always wait for the updated list; I have added some checks so they will wait for the URL update.

Sometimes, (1 out of 5 tests?) the test fails with the console error: "Minified React error #418; visit https://reactjs.org/docs/error-decoder.html?invariant=418 for the full message or use the non-minified dev environment for full errors and additional helpful warnings."

This happens randomly on different pages. I never get this error locally, and the app will still work when this happens. Since this does not impact the user, I ignored this error.

# out of scope

I do not add enrichments; after discussing with Sjors, we decided that we could start adding tests for this, but I will do this separately.

### CODE REVIEW

**Review by sdevalk — COMMENTED** [review]
Nice! I've added some suggestions.

**Comment by barbarah** [comment]
After the merge with `main`, there are new problems with Cypress config importing from packages that still need to be converted to ESM. I created a pull request that fixes this problem.

File: apps/researcher/cypress/e2e/community.cy.ts [hunk 1]
```diff
@@ -42,3 +42,60 @@ describe('Community details page', () => {
...
+    cy.visit('/en/communities', {
+      failOnStatusCode: false,
+    });
+
+    cy.getBySel('add-community').click();
+    // I can't add test ids to the modal, so I'm using the h1
+    cy.get('h1').contains('Create Community');
+  });
+
+  it('can find my community with the "Show only my communities" toggle', () => {
```
Comment (sdevalk) [inline apps/researcher/cypress/e2e/community.cy.ts:61]: Active voice, 'it finds my community...'?

File: apps/researcher/cypress/e2e/community.cy.ts [hunk 2]
```diff
@@ -42,3 +42,60 @@ describe('Community details page', () => {
...
+    cy.getBySel('community-item-name').should('have.length', 1);
+  });
+});
+
+describe('Community details page logged in', () => {
+  beforeEach(() => {
+    cy.session('signed-in', () => cy.signIn());
+  });
+
+  it('can edit my community', () => {
```
Comment (sdevalk) [inline apps/researcher/cypress/e2e/community.cy.ts:76]: Active voice, 'it edits...'?

File: apps/researcher/cypress/e2e/community.cy.ts [hunk 3]
```diff
@@ -42,3 +42,60 @@ describe('Community details page', () => {
...
+      .type(
+        `This community is used for end-to-end testing; please do not remove or use this community. Unique Identifier: ${uniqueIdentifier}`
+      );
+    cy.getBySel('save-button').click();
+
+    cy.getBySel('notification').should('exist');
+    cy.contains(uniqueIdentifier).should('exist');
+  });
+
+  it('can open the manage user modal', () => {
```
Comment (sdevalk) [inline apps/researcher/cypress/e2e/community.cy.ts:94]: Active voice, 'it opens...'?

File: apps/researcher/cypress/e2e/object-list.cy.ts [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+});
+
+describe('Object list page logged in', () => {
+  beforeEach(() => {
+    cy.task('resetDb');
+    cy.task('createEmptyList').as('listId');
+    cy.session('signed-in', () => cy.signIn());
+  });
+
+  it('can edit the list name and description', function () {
```
Comment (sdevalk) [inline apps/researcher/cypress/e2e/object-list.cy.ts:61]: Active voice, 'it edits...'?

File: apps/researcher/cypress/e2e/object-list.cy.ts [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    );
+
+    cy.getBySel('edit-list-button').click();
+    cy.get('#name').type(' Edited');
+    cy.getBySel('save-button').click();
+    cy.getBySel('notification').should('exist');
+    cy.contains('Edited').should('exist');
+  });
+
+  it('can delete the object list', function () {
```
Comment (sdevalk) [inline apps/researcher/cypress/e2e/object-list.cy.ts:76]: Active voice, 'it deletes...'?

File: apps/researcher/cypress/e2e/object-list.cy.ts [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+      `/en/communities${Cypress.env('TEST_COMMUNITY_SLUG')}/${this.listId}`,
+      {
+        failOnStatusCode: false,
+      }
+    );
+
+    cy.getBySel('no-entity').should('exist');
+  });
+
+  it('can delete an object from the list', function () {
```
Comment (sdevalk) [inline apps/researcher/cypress/e2e/object-list.cy.ts:103]: Active voice, 'it deletes...'?

File: apps/researcher/cypress/e2e/object.cy.ts [hunk 7]
```diff
@@ -60,3 +60,43 @@ describe('Object details page', () => {
...
+        {
+          failOnStatusCode: false,
+        }
+      );
+
+      cy.getBySel('object-card').should('have.length', 1);
+    });
+  });
+
+  it("can open the 'add enrichment' form", () => {
```
Comment (sdevalk) [inline apps/researcher/cypress/e2e/object.cy.ts:92]: Active voice, 'it opens...'?

File: apps/researcher/cypress/e2e/community.cy.ts [hunk 8]
```diff
@@ -42,3 +42,60 @@ describe('Community details page', () => {
...
+    cy.session('signed-in', () => cy.signIn());
+  });
+
+  it("opens the 'add community' modal", () => {
+    cy.visit('/en/communities', {
+      failOnStatusCode: false,
+    });
+
+    cy.getBySel('add-community').click();
+    // I can't add test ids to the modal, so I'm using the h1
```
Comment (sdevalk) [inline apps/researcher/cypress/e2e/community.cy.ts:57]: Minor: who is 'I' and "I'm"? Perhaps use a general 'we' or an abstract "It's not possible to add test IDs to the model, so we're using the h1 instead"?

**Review by sdevalk — APPROVED** [review]
Looking good! I've added some minor style remarks.

**Pull request merged**